src/main/java/io/kamax/matrix/client/AMatrixHttpClient.java

```diff
@@ -75,14 +81,27 @@ public _MatrixHomeserver getHomeserver() {
 
     @Override
     public String getAccessToken() {
-        return context.getToken();
+        return context.getToken()
+                .orElseThrow(() -> new IllegalStateException("This method can only be used with a valid token."));
```

Comment:
The CS API don't require login everywhere, like fetching media or getting the list of supported protocol version. This cannot be made mandatory.